### ISSUE TITLE
Keyboard commands (other than opening current file) don't activate or focus the new popover

### ISSUE THREAD

**Issue by pjeby** [nothingislost/obsidian-hover-editor#22]
This happens regardless of the "auto focus" setting, 

(Also, opening a popover for the current link does not create a new file when it doesn't exist.)

**Comment by nothingislost** [comment]
I'll have this fixed in the next release except for the fact that it doesn't seem like empty pane views allow for cursor/selection focus. They'll still become the active pane though which should be good enough for most purposes.

**Comment by pjeby** [comment]
This doesn't appear to be fixed: the "open current file in new popover" and "open link under cursor in new popover" don't set the popover to active at all, and "open new popover" sets the CSS such that it shows active, but trying to do anything with it that opens a modal (e.g. open or the command palette) ends up targeting the recently active pane instead.

These seem to perhaps be two different problems?

The second problem appears to be that when a dialog closes, it tries to restore the focus and selection, and the previous focus for an "empty" view is the body, and that's apparently what triggers going back to the recently active pane.  As a workaround, I did the following in the console (which should only be done to a fully loaded "empty" view (i.e., if the leaf's `.view.getViewType()==="empty"`):

```js
app.workspace.activeLeaf.view.contentEl.tabIndex = -1
app.workspace.activeLeaf.view.contentEl.focus()
```

That worked to focus the empty view, which then meant that dialog closing didn't refocus to the main panes.

This exposed another problem, which is that this line:

https://github.com/nothingislost/obsidian-hover-editor/blob/3899a9ca2ebb928f1929dbb7a9cde8db822b724d/src/leaf.ts#L125

doesn't check for a null `.state`.  The actual type of OpenViewState has both eState and state as optional: they can be non-existent.  Notably, the call made by the quick switcher to openFile() just passes `{active: true}` as the openState.  (Which your commands' calls to openFile and openLinkText should also be doing, so the leaf will activate.)

I've created #24 to implement the above, and the commands all now work as one would expect: all commands will focus the created pane, even if it's empty, and following a broken link will create the file.  I had to add an onShowCallback facility to HoverEditor, as part of the reason that new popovers weren't focusing is that they're not mounted in the DOM until show() runs.

**Comment by csss1234** [comment]
I'm still having this problem, unable to trigger with "open under cursor"

**Comment by pjeby** [comment]
This should be fixed in HE 0.11.2, please try it and let us know.  Thanks!

**Comment by csss1234** [comment]
> This should be fixed in HE 0.11.2, please try it and let us know. Thanks!

Updated. Still not working for me

**Comment by pjeby** [comment]
What version of Obsidian (including installer version) and what version of HE?  Please double-check your HE version to make sure it is actually 0.11.2.

**Comment by csss1234** [comment]
Sorry I should have done that before
HE .11.2
Obs: .15.6 with installer version .14.6

**Comment by pjeby** [comment]
And you're using "Open link under cursor in new Hover Editor" and nothing is happening?  Are you in the main window or a popout?  If you use other commands like "Follow link under cursor" or "Open link under cursor in a new pane" at the *same cursor position* where the hover editor command isn't working, do they work?

**Comment by csss1234** [comment]
Ohhh, I figured it out. Sorry I misunderstood. I thought by cursor, it meant "_mouse_ pointer/cursor". 

I guess because thats how the core page preview works (hover the mouse pointer over anything and hit command). But this "follow/open link under cursor" means the _text_ cursor. 

Would be cool if the feature worked as I had expected, because I'm always triggering hover editor when I just want to run any other ⌘ command on mac. I know this is discussed elsewhere here. I figured if I could get HE to trigger "open link under cursor" with alt or option + key, then I could maybe disable it being triggered by ⌘.
